src/main/drivers/accgyro/accgyro.h

```diff
@@ -49,7 +49,7 @@ typedef struct gyroDev_s {
     sensorGyroInterruptStatusFuncPtr intStatusFn;
     sensorGyroUpdateFuncPtr updateFn;
     extiCallbackRec_t exti;
-    busDevice_t bus;
+    busDevice_t * dev;
```

Comment:
Why is this a pointer?

Reply:
Because `busDevice_t` object may be shared between gyro and acc (and compass).

Reply:
Yes, it can me shared. But it is small and not often shared. I think it is better to duplicate.

Reply:
@martinbudden the reason to share the object is that we do device detection (and most of the configuration) only in gyro driver. Acc driver needs to know if gyro device was detected and initialized - shared `scratchpad` field of the `busDevice_t` is used for that.

Reply:
The way this is currently handled is:
1. acc init occurs after gyro init
2. acc init copies gyro busDevice_t data.

It's straightforward, has no potential for errors. Not also this works for dual gyro systems in Betaflight.

Reply:
I never likes the part of initialization that was copying around data between `gyro_t` and `acc_t`. High-level code needs to know nothing about how devices operate. All hardware sharing issues should be solved at low level. 

High-level code needs to do two things: 1) call `accWhateverDetect()` and 2) call `acc->init()` on detected device. 

If we were sticking to pragmatic approach - we would still be flying Baseflight code :smile: